hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -896,6 +896,14 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// If cluster needs scheduling and is not scheduled, then wait until it is
+	if needsScheduling(hcluster) {
```

Comment:
why are we making the hc aware of management side scheduling? Why this vs letting the workflow try and organically fail.
If we move forward with this we'll want to signal this state in a condition. We could probably reuse ReconciliationActive with reason: unschedulable or so.

Reply:
This is our scheduling that we are gating on. Our scheduler controller needs to find 2 nodes that are available for this hosted cluster, taint them, and then allow the hc to proceed.

One other thought that occurred to me is that maybe this is worth adding API fields and such, given that we want to report status on it and everything. Thoughts?